### DISCUSSION TITLE
Do not install "devDependency" of the project dependencies

### DISCUSSION THREAD

**Discussion by xr0master** [npm/feedback#669] [CLI]
Many teams use different tools to develop their packages. The size of these tools can be enormous. At the same time, in our projects we use their finished products, we do not develop these packages but only use them.

Therefore, only product dependencies should be installed when a package is added to a project. This can save up to 0.5 GB on average for each project! This should also reduce the number of vulnerable packages on the computer.

What do you think about this?

Edit:
Added screenshots to visually see the issue. The `firebase-tools` are used as an example.
The `@google/events` package is installed.

<img width="457" alt="image" src="https://user-images.githubusercontent.com/8962340/163463298-cf38e203-f20c-4602-81da-073bef449107.png">

However, in the `firebase-tools` package.json, the `@google/events` is marked as "devDependency".

<img width="301" alt="image" src="https://user-images.githubusercontent.com/8962340/163463397-9be5edd6-fa0a-4d7c-b6e2-ad2e6cbc3b10.png">

The `firebase-tools` in our project package.json

<img width="285" alt="image" src="https://user-images.githubusercontent.com/8962340/163463631-bedb69a2-793f-40e8-8ae3-31d7e1df84fb.png">

**Comment by ljharb** [comment]
That's already how it works. Dev deps are only installed from the top-level project.

**Comment by xr0master** [comment]
@ljharb unfortunately, it's not like this. For example, the firebase-tools has in node_modules the puppeteer. And they have this package added in "devDependencies".

**Reply by ljharb** [reply]
I'm quite certain this already is how it works. You can run `npm explain puppeteer` to see why it's there, though.

**Reply by xr0master** [reply]
I don't know where you get that confidence from, but it definitely doesn't work that way :)
At least in npm version 8.

<img width="407" alt="image" src="https://user-images.githubusercontent.com/8962340/163460687-d3ecbc66-199a-4a12-b135-e4c63f2ef4fa.png">

**Reply by ljharb** [reply]
I see it's there, but it shouldn't be by default. What version of npm are you using? Do you have any env vars set that might set "--all", or "--depth"?

**Reply by xr0master** [reply]
the npm version is `8.5.5`. There are no env variables that I could be aware of.

**Reply by xr0master** [reply]
I run the `npm config ls -l`: `depth = null` and `all = false`

**Reply by ljharb** [reply]
In that case, I'd suggest filing a bug on https://github.com/npm/cli :-)
